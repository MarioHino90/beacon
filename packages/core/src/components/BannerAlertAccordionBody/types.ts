export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
