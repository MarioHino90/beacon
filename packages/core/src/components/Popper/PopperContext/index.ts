export { PopperContext } from "./PopperContext";

export {
  PopperContextProvider,
  placementTypes,
  usePopperContext,
} from "./PopperContext";
