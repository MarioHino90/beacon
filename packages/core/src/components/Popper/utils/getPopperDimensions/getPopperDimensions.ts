import { normalizeDimension } from "../normalizeDimension";

/**
 *
 * @param popperRef
 */
export const getPopperDimensions = (
  popperRef: React.RefObject<HTMLElement>
): [number, number] => {
  // get the dimensions of the Popper element
  const elementHeight = normalizeDimension(
    popperRef.current?.getBoundingClientRect().height || 0
  );
  const elementWidth = normalizeDimension(
    popperRef.current?.getBoundingClientRect().width || 0
  );

  return [elementHeight, elementWidth];
};
