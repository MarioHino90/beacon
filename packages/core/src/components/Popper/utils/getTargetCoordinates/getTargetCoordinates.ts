import { RefObject } from "react";

import { TargetCoordinates } from "../../types";

/**
 * Get coordinates for the target element on the DOM
 * @param element an HTML element instance
 * @param portalRef ref of Portal inside which Popper will be rendered
 */
export const getTargetCoordinates = <P extends HTMLElement = HTMLDivElement>(
  element: HTMLElement,
  portalRef?: RefObject<P>
): TargetCoordinates => {
  // calculate any offsets based on scroll and/or page layout behavior
  const pageLeftOffset =
    window.pageXOffset || document.documentElement.scrollLeft;
  const pageTopOffset =
    window.pageYOffset || document.documentElement.scrollTop;

  // get the coordinates of the target element
  const rect = element.getBoundingClientRect();

  // if portalRef.current is defined, then calculate element's
  // positioning relative to the portal parent
  if (portalRef?.current) {
    // get portalRef's parent bounding rect
    const parentRect = portalRef.current.getBoundingClientRect();

    // return relative coordinates
    return {
      top: rect.top - parentRect.top + pageTopOffset,
      right: rect.width,
      bottom: rect.height,
      left: rect.left - parentRect.left + pageLeftOffset,
    };
  }

  // if portalRef.current isn't defined,
  // then return the absolute coords
  return {
    top: rect.top + pageTopOffset,
    right: rect.width,
    bottom: rect.height,
    left: rect.left + pageLeftOffset,
  };
};
