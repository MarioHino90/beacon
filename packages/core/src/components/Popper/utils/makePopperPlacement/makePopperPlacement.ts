import { PopperContextInterface, PopperPlacement } from "../../types";
import { getTargetCoordinates } from "../getTargetCoordinates";
import { getPopperDimensions } from "../getPopperDimensions";
import { getAutoPlacement } from "../getAutoPlacement";

export const makePopperPlacement = (
  element: HTMLElement | null,
  popperRef: React.RefObject<HTMLElement>,
  {
    defaultPlacement,
    distance = 0,
    interactive,
    placement,
    portalRef,
    skidding = 0,
  }: Partial<PopperContextInterface>
): PopperPlacement => {
  // if element is not defined, short-circuit
  if (!element) {
    return {
      left: 0,
      top: 0,
      activePlacement: "bottom",
    };
  }

  // calculate offset to ensure that overlap of popper when interactive
  const interactiveOffset = interactive ? -10 : 0;

  // get coordinates of target on the DOM
  const targetCoordinates = getTargetCoordinates(element, portalRef);

  // get the dimensions of the Popper element
  const [popperHeight, popperWidth] = getPopperDimensions(popperRef);

  // auto-placement calculations, if 'placement' is 'auto'
  const activePlacement =
    placement !== "auto"
      ? placement
      : getAutoPlacement(
          element,
          { popperHeight, popperWidth },
          defaultPlacement
        );

  // break up coords for easier readability
  const { top, right, bottom, left } = targetCoordinates;

  // determine final placement of Popper
  switch (activePlacement) {
    case "bottom":
      return {
        left: left + (right - popperWidth) / 2 + skidding,
        top: top + bottom + interactiveOffset + distance,
        activePlacement,
      };
    case "bottom-start":
      return {
        left: left + skidding + interactiveOffset,
        top: top + bottom + interactiveOffset + distance,
        activePlacement,
      };
    case "bottom-end":
      return {
        left: left - (popperWidth - right) + skidding - interactiveOffset,
        top: top + bottom + interactiveOffset + distance,
        activePlacement,
      };
    case "top":
      return {
        left: left + (right - popperWidth) / 2 + skidding,
        top: top - popperHeight - interactiveOffset - distance,
        activePlacement,
      };
    case "top-start":
      return {
        left: left + skidding + interactiveOffset,
        top: top - popperHeight - interactiveOffset - distance,
        activePlacement,
      };
    case "top-end":
      return {
        left: left - (popperWidth - right) + skidding - interactiveOffset,
        top: top - popperHeight - interactiveOffset - distance,
        activePlacement,
      };
    case "right":
      return {
        left: left + right + interactiveOffset + distance,
        top: top + (bottom - popperHeight) / 2 + skidding,
        activePlacement,
      };
    case "right-start":
      return {
        left: left + right + interactiveOffset + distance,
        top: top + skidding + interactiveOffset,
        activePlacement,
      };
    case "right-end":
      return {
        left: left + right + interactiveOffset + distance,
        top: top - (popperHeight - bottom) + skidding - interactiveOffset,
        activePlacement,
      };
    case "left":
      return {
        left: left - popperWidth - interactiveOffset - distance,
        top: top + bottom / 2 - popperHeight / 2 + skidding,
        activePlacement,
      };
    case "left-start":
      return {
        left: left - popperWidth - interactiveOffset - distance,
        top: top + skidding + interactiveOffset,
        activePlacement,
      };
    case "left-end":
      return {
        left: left - popperWidth - interactiveOffset - distance,
        top: top - (popperHeight - bottom) + skidding - interactiveOffset,
        activePlacement,
      };
    default:
      return {
        left: left + (right - popperWidth) / 2 + skidding,
        top: top - popperHeight - interactiveOffset - distance,
        activePlacement: "bottom",
      };
  }
};
