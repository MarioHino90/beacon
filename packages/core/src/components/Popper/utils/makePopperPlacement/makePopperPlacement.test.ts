import { createRef } from "react";
import faker from "@faker-js/faker";

import { placementTypes } from "../../PopperContext";
import { getTargetCoordinates } from "../getTargetCoordinates";
import { getPopperDimensions } from "../getPopperDimensions";
import { PlacementType } from "../../types";

import { makePopperPlacement } from "./makePopperPlacement";

faker.seed(20200813);

jest.mock("../getTargetCoordinates", () => ({
  getTargetCoordinates: jest.fn(() => ({
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  })),
}));

jest.mock("../getPopperDimensions", () => ({
  getPopperDimensions: jest.fn(() => [0, 0]),
}));

describe(`makePopperPlacement`, () => {
  const fakeElement = document.createElement("div");
  const fakeRef = createRef<HTMLDivElement>();

  it(`should match snapshot when element is NOT defined`, () => {
    expect(makePopperPlacement(null, fakeRef, {})).toMatchSnapshot();
  });

  describe.each([
    ...placementTypes,
    "some-other-value" as unknown as PlacementType,
  ])(
    `when 'placement' is '%s' and 'element' is defined`,
    (placement: PlacementType) => {
      beforeEach(() => {
        (getTargetCoordinates as jest.Mock).mockImplementationOnce(() => [
          faker.datatype.number(),
          faker.datatype.number(),
          faker.datatype.number(),
          faker.datatype.number(),
        ]);

        (getPopperDimensions as jest.Mock).mockImplementationOnce(() => [
          faker.datatype.number(),
          faker.datatype.number(),
        ]);
      });

      it(`should match snapshot`, () => {
        expect(
          makePopperPlacement(fakeElement, fakeRef, {
            placement,
          })
        ).toMatchSnapshot();
      });

      it(`should match snapshot when 'interactive' is true`, () => {
        expect(
          makePopperPlacement(fakeElement, fakeRef, {
            placement,
            interactive: true,
          })
        ).toMatchSnapshot();
      });

      it.each([0, 5, 10])(
        `should match snapshot when 'skidding' is '%s'`,
        (skidding) => {
          expect(
            makePopperPlacement(fakeElement, fakeRef, {
              placement,
              skidding,
            })
          ).toMatchSnapshot();
        }
      );

      it.each([0, 5, 10])(
        `should match snapshot when 'distance' is '%s'`,
        (distance) => {
          expect(
            makePopperPlacement(fakeElement, fakeRef, {
              placement,
              distance,
            })
          ).toMatchSnapshot();
        }
      );
    }
  );
});
