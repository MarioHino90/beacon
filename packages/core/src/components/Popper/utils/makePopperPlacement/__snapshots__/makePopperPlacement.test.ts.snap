// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makePopperPlacement should match snapshot when element is NOT defined 1`] = `
{
  "activePlacement": "bottom",
  "left": 0,
  "top": 0,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'auto' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-end' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "bottom-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'bottom-start' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "bottom-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "left",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-end' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "left-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'left-start' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "left-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "right",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-end' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "right-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'right-start' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "right-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'some-other-value' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "bottom",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "top",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-end' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "top-end",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'distance' is '0' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'distance' is '5' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'distance' is '10' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'interactive' is true 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'skidding' is '0' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'skidding' is '5' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;

exports[`makePopperPlacement when 'placement' is 'top-start' and 'element' is defined should match snapshot when 'skidding' is '10' 1`] = `
{
  "activePlacement": "top-start",
  "left": NaN,
  "top": NaN,
}
`;
